cookie情報が古く、ログインの期限が切れた場合にpydanticのバリデーションエラーになる

## 詳細
`model_post_init` 関数内の `r = self.session.get(TalkURL.HOST, allow_redirects=False)` がログイン期限切れでも200を返す。
なので、bot情報取得APIを叩いた時に、不正なレスポンスで `GetMyInfoResponse` をパースしようとしている。
そこでバリデーションエラーが起きている

## 対策・修正

bot情報取得APIを叩くようにしてバリデーションエラーが起きれば、再ログインする処理に変更

## 実行ログ
```bash
(venv) [17:51:16] ~/github.com/line-works-sdk (develop) 
$ make run
python main.py
2024/12/13 17:51:22 [INFO] __main__:37 hello.
Traceback (most recent call last):
  File "/Users/macuser/github.com/line-works-sdk/main.py", line 39, in <module>
    works = LineWorks(works_id=WORKS_ID, password=PASSWORD)
  File "/Users/macuser/github.com/line-works-sdk/venv/lib/python3.13/site-packages/pydantic/main.py", line 214, in __init__
    validated_self = self.__pydantic_validator__.validate_python(data, self_instance=self)
pydantic_core._pydantic_core.ValidationError: 30 validation errors for LineWorks
tenantId
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
domainId
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
contactNo
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
readOnly
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
tempId
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
name
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
i18nName
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
i18nNames
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
photos
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
organizations
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
emails
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
telephones
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
messengers
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
position
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
department
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
location
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
important
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
executive
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
photoHash
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
worksAt
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
accessLimit
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
userPhotoModify
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
userAbsenceModify
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
organization
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
groups
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
worksServices
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
customFields
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
profileStatuses
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
profileStatusesV2
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
instance
  Field required [type=missing, input_value={'code': 401, 'message': ...ed', 'planError': False}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.10/v/missing
make: *** [run] Error 1
```

--- a/.github/workflows/lint_python.yaml
+++ b/.github/workflows/lint_python.yaml
@@ -23,7 +23,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.10", "3.11", "3.12"]
+        python-version: ["3.10", "3.11", "3.12", "3.13"]
 
     steps:
       - uses: actions/checkout@v3

--- a/.github/workflows/publish_pypi.yml
+++ b/.github/workflows/publish_pypi.yml
@@ -8,6 +8,12 @@ on:
     branches:
       - develop
   pull_request:
+    paths:
+      - "**.py"
+      - pyproject.toml
+      - requirements-dev.txt
+      - requirements.txt
+      - setup.cfg
 
 jobs:
   setup:

--- a/.github/workflows/publish_pypi.yml
+++ b/.github/workflows/publish_pypi.yml
@@ -67,7 +67,7 @@ jobs:
           password: ${{ secrets.PYPI_TOKEN }}
 
   test-pypi:
-    if: ${{ github.event_name == 'pull_request' }}
+    if: ${{ github.event_name == 'pull_request' && github.actor != 'dependabot[bot]' }}
     needs: setup
     runs-on: ubuntu-latest
     environment:

--- a/line_works/client.py
+++ b/line_works/client.py
@@ -2,10 +2,10 @@ import json
 from os import makedirs
 from os.path import exists
 from os.path import join as path_join
-from typing import Any, Optional
+from typing import Any
 
-from pydantic import BaseModel, Field
-from requests import HTTPError, JSONDecodeError, Response, Session
+from pydantic import BaseModel, Field, ValidationError
+from requests import HTTPError, JSONDecodeError, Session
 
 from line_works import config
 from line_works.decorator import save_cookie
@@ -46,19 +46,19 @@ class LineWorks(BaseModel):
         makedirs(self.session_dir, exist_ok=True)
         self.session.headers.update(config.HEADERS)
 
-        r: Optional[Response] = None
-
         if exists(self.cookie_path):
             # login with cookie
             with open(self.cookie_path) as j:
                 c = json.load(j)
             self.session.cookies.update(c)
-            r = self.session.get(TalkURL.HOST, allow_redirects=False)
 
-        if not r or r.status_code != 200:
+        try:
+            my_info = self.get_my_info()
+        except ValidationError as _:
+            self.session.cookies.clear()
             self.login_with_id()
+            my_info = self.get_my_info()
 
-        my_info = self.get_my_info()
         self.tenant_id = my_info.tenant_id
         self.domain_id = my_info.domain_id
         self.contact_no = my_info.contact_no

